### DISCUSSION TITLE
Next.js MDX doesn't work with Remark ESM plugins

### DISCUSSION THREAD

**Discussion by leap0x7b** [vercel/next.js#29315] [Help]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

16.9.1

### What browser are you using?

Chrome

### What operating system are you using?

Linux

### How are you deploying your application?

`next` (start in development environment)

### Describe the Bug

If you use a Remark 11 ESM plugin, it won't be applied. This bug occurred to me when I'm making a Next.js project that uses MDX

### Expected Behavior

It should apply the plugin

### To Reproduce

1. Install a Remark 11 ESM plugin
2. Add the plugin using `import()` in `remarkPlugins`
3. Do either `next start` for production environment or `next` for development environment

**Comment by leap0x7b** [comment]
Could you guys please help

**Comment by xxczaki** [comment]
Don't expect to get much help quickly when your issue is moved to discussions. I was dealing with the same issue today and the solution is to use Next.js 12 and `next.config.mjs` (instead of `.js`), like so:

```js
import math from 'remark-math';
import katex from 'rehype-katex';
import withMDX from '@next/mdx';

const nextConfig = {};

export default withMDX({
	extension: /\.mdx?$/,
	options: {
		providerImportSource: '@mdx-js/react',
		remarkPlugins: [
			math,
		],
		rehypePlugins: [
			katex,
		],
	},
})(nextConfig);
```
